aws-xray-agent/src/main/java/com/amazonaws/xray/agent/runtime/handlers/downstream/SqlHandler.java

```diff
@@ -68,7 +76,7 @@ public void handleRequest(Event event) {
     @Override
     public void handleResponse(Event event) {
         // If this SQL request is being ignored, we should also ignore the response
-        if (decrementSqlTransactionCount() > 0) {
+        if (decrementSqlTransactionCount() > 0 || event == null) {
```

Comment:
Why event is null, is it caused by mock?

Reply:
Yep, I discovered this while mocking. In practice it should never be null, but I figured it couldn't hurt to add the check.

Reply:
Adding `event == null` involves testing logic to code, it is not the best practice unless there is no better way.